src/cs/LionWeb.Integration.WebSocket.Tests/Client/ContainmentClientTests.cs

```diff
@@ -29,7 +33,78 @@ public void AddChild()
 
         AssertEquals(aPartition, bPartition);
     }
+    
+    /// <summary>
+    /// Added child is a (complex) subtree
+    /// </summary>
+    [Test]
+    public void AddChild_AsASubtree()
+    {
+        var subTree = new LinkTestConcept("subtree")
+        {
+            Name = "containment_0_1",
+            Containment_0_1 = new LinkTestConcept("containment_0_1"),
+            Containment_0_n = new List<LinkTestConcept> { new("child1"), new("child2") }
+        };
+
+        aPartition.Containment_0_1 = subTree;
+        bClient.WaitForReplies(1);
+
+        AssertEquals(aPartition, bPartition);
+    }
+
+    /// <summary>
+    /// Node in an added subtree has a reference to already existing node
+    /// </summary>
+    [Test]
+    public void AddChild_NodeInAddedSubtreeHasAReferenceToAlreadyExistingNodes()
+    {
+        aPartition.Containment_1 = new LinkTestConcept("referenced-child");
+        bClient.WaitForReplies(1);
+
+        AssertEquals(aPartition, bPartition);
+
+        bPartition.Containment_1 = new LinkTestConcept("subtree")
+        {
+            Reference_1 = bPartition.Containment_1
+        };
+        aClient.WaitForReplies(1);
+
+        AssertEquals(aPartition, bPartition);
+    }
+    
+    /// <summary>
+    /// Already existing node has a reference to a node in an added subtree
+    /// </summary>
+    [Test]
+    public void AddChild_AlreadyExistingNodeHasAReferenceToNodeInAddedSubtree()
+    {
+        aPartition.Containment_0_1 = new LinkTestConcept("existing-subtree")
+        {
+            Containment_0_1 = new LinkTestConcept("containment")
+        };
+        bClient.WaitForReplies(1);
+
+        AssertEquals(aPartition, bPartition);
 
+        aPartition.Containment_1 = new LinkTestConcept("added-node");
+        bClient.WaitForReplies(1);
+        
+        AssertEquals(aPartition, bPartition);
+        
+        // This works, but reference link is established after the nodes are added.
+        // One would expect reference from a child is set to the node before the new node
```

Comment:
Why? If I get the comment right, somewhere in the middle of the process inside delta processing the reference is not set, but as soon as the user sees it we're good?